plutarch-extra/Plutarch/Extra/ErrorCode.hs

```diff
@@ -0,0 +1,71 @@
+module Plutarch.Extra.ErrorCode (errorTagReplacer, codifyStrings) where
+
+import Data.List (nub)
+import Data.Maybe (fromMaybe)
+import Data.Text (Text)
+import Data.Text qualified as T
+import Data.Tuple (swap)
+import Plutarch.Script (Script (Script))
+import PlutusCore (DefaultUni (..))
+import Universe (Some (Some), ValueOf (ValueOf))
+import UntypedPlutusCore qualified as UPLC
+
+-- | Replace any string that is tagged with "ERR:" prefix will get replaced by a unique integer id
+errorTagReplacer :: Int -> Text -> Maybe Text
+errorTagReplacer n s =
+  if T.isPrefixOf "ERR:" s
+    then Just $ "CODE:" <> T.pack (show n)
+    else Nothing
+
+-- | Replace strings satisfying the predicate with provided replacement
+codifyStrings :: (Int -> Text -> Maybe Text) -> Script -> (Script, [(Text, Text)])
+codifyStrings codify (Script (UPLC.Program ann version terms)) =
+  let
+    immap :: forall a b. (Int -> a -> Maybe b) -> [a] -> [(a, b)]
+    immap f xs = go 0 xs
```

Comment:
Were you looking for [this](https://hackage.haskell.org/package/witherable-0.5/docs/Witherable.html#t:WitherableWithIndex) perhaps?